db/model/subject.js

```diff
@@ -326,10 +313,20 @@ module.exports = function subject(seq, dataTypes) {
 
         if (isSubjectUnpublished) {
           // Treat unpublishing a subject as a "delete" event.
-          common.publishChange(inst, eventName.del);
+          publishObject(inst, eventName.del);
         } else if (isSubjectPublished) {
           // Treat publishing a subject as an "add" event.
-          common.publishChange(inst, eventName.add);
+          publishObject(inst, eventName.add);
+        } else if (inst.isPublished && inst.changed('absolutePath')) {
+          /*
+           * When an absolutePath is changed, send a subject delete event with
+           * the old subject instance, followed by a subject add event with the
+           * new subject instance
+           */
+          publishObject(inst._previousDataValues, eventName.del,
+            changedKeys, ignoreAttributes);
+          publishObject(inst, eventName.add, changedKeys,
+              ignoreAttributes);
```

Comment:
L320 to L329 were the only changes required. The rest is just clean. (I went down a wrong path before realizing this)